// react-native-fb-collage/built/fb-collage.style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = (function (width, height, borderRadius, spacing) {
    var space = spacing || width / 29.75, imageWidth = ((width / 2) / 2) - (2 * space), imageHeight = (height / 2) - (1.5 * space);
    return {
        VIEW: {
            width: width,
            height: height,
            paddingRight: space,
            paddingEnd: space,
            paddingTop: space,
            flexDirection: 'row',
            alignSelf: 'center'
        },
        FLEX: {
            flex: 1
        },
        IMAGE: function (length) {
            return (length < 2
                ? {
                    flex: 1
                }
                : {
                    flex: length > 3 ? 0 : 1,
                    width: imageWidth,
                    height: imageHeight
                });
        },
        CONTAINER: function (length) {
            return (length < 2
                ? {
                    flex: 1
                }
                : {
                    flexDirection: 'column',
                    flexWrap: 'wrap'
                });
        },
        BUTTON: {
            flex: 1,
            borderRadius: borderRadius,
            marginBottom: space,
            marginLeft: space,
            marginStart: space,
            overflow: 'hidden'
        },
        OVERLAY: {
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center',
            backgroundColor: '#0000009d'
        },
        TEXT: {
            fontSize: 24,
            lineHeight: 28,
            fontWeight: '600',
            color: 'white'
        }
    };
});
